### PR TITLE
[FEATURE!] Disable Conjunction for Certain Operators

### PR DESCRIPTION
**Pull request by arleytm**
This PR enables conjunction for specific operators in the CAP GraphQL Adapter, allowing for combining multiple conditions in queries. The proposed configuration has been refined based on team discussions and stakeholders and incorporates valuable input from all members.

### Updated the configuration to enable conjunction for the following operators
  - `ne` (Not Equal)
  - `contains`

### Disabled conjunction for the following operators
  - `eq` (Equal)
  - `gt` (Greater Than)
  - `ge` (Greater Than or Equal)
  - `le` (Less Than or Equal)
  - `lt` (Less Than)
  - `startswith`
  - `endswith`

### Sample query using `contains` operator as a filter

```
query MyQuery {
  AdminService {
    Books(filter: {title: {contains: ["Wuthering", "Heights"]}}) {
      nodes {
        title
      }
    }
  }
}
```

## Reasoning
- Enabling conjunction for `ne` and `contains` allows combining multiple inequality and string condition in queries, providing greater flexibility.
- Disabling conjunction for other operators simplifies the query evaluation process. When multiple values are provided for operators such as `eq`, `gt`, `ge`, `le`, `lt`, only the lowest or highest value in the list will be considered, making the other values without effect. By disabling conjunction for these operators, the query can be simplified to a single value without evaluating additional values, enhancing query performance and reducing complexity.

### CODE REVIEW

**Review by schwma — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (arleytm) [reply CHANGELOG.md:23]: > on the same field

👍

As for the `+` operator, adding it would actually violate the [MD004/ul-style](https://github.com/DavidAnson/markdownlint/blob/v0.28.2/doc/md004.md) rule.

Comment (schwma) [reply CHANGELOG.md]: Did not know about this. But doesn't the last example explain that it is ok as long as the symbols are consistent for each indentation level?

I think we should use `+` for indentations as that is what we have consistently done in changelogs of other `cds` packages.

Comment (arleytm) [reply CHANGELOG.md]: I just received a warning in VSCode. So, stick with the `+` operator.

MD004:ul-style

**Review by schwma — APPROVED** [review]
Thanks for the changes, LGTM 👍 

Maybe we should still wait for @johannes-vogel approval of the proposal before merging

**Pull request merged**
